### PR TITLE
Book model

### PR DESCRIPTION
**Pull request by sebasdeldi**
**Description:**
- Adds Book model and attributes

**Card:**
https://trello.com/c/sH5Ubcev/14-crear-modelo-de-book

### CODE REVIEW

File: src/main/java/com/wolox/training/models/Book.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Book {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private Long id;
+  private String genre;
```
Comment (Gianni-Nicolas) [inline src/main/java/com/wolox/training/models/Book.java:19]: Add the @Colum annotation for the genre attribute.

File: src/main/java/com/wolox/training/models/Book.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  @Column(nullable=false)
+  private String publisher;
+
+  @Column(nullable=false)
+  private String year;
+
+  @Column(nullable=false)
+  private  Integer pages;
+
+  @Column(nullable=false)
```
Comment (Gianni-Nicolas) [inline src/main/java/com/wolox/training/models/Book.java]: To unique values ​​such as isbn add the property "unique = true" as follows:

@Column (nullable = false, unique = true).

File: src/main/java/com/wolox/training/models/Book.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  private  String isbn;
+
```
Comment (Gianni-Nicolas) [inline src/main/java/com/wolox/training/models/Book.java:44]: Add the empty constructor to the class. This is necessary for Hibernate to initialize the entity.

File: src/main/java/com/wolox/training/models/Book.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Book {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (Gianni-Nicolas) [inline src/main/java/com/wolox/training/models/Book.java]: In Postgres when GenerationType.AUTO is used, it generates an incremental IDs shared between the schema tables. This happens only for this database engine. To generate independent IDs for each table use GenerationType.SEQUENCE.

  ```
  @GeneratedValue (strategy = GenerationType.SEQUENCE, generator = "BOOK_SEQ")
    @SequenceGenerator(name = "BOOK_SEQ", sequenceName = "BOOK_SEQ")
```

File: src/main/java/com/wolox/training/models/Book.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+  public void setIsbn(String isbn) {
+    this.isbn = isbn;
+  }
+
+  public Long getId() {
+    return id;
+  }
+
+  public void setId(Long id) {
```
Comment (Gianni-Nicolas) [inline src/main/java/com/wolox/training/models/Book.java]: The IDs generated by the entity, thanks to the @GeneratedValue annotation, make it unnecessary to include the method "public void setId (Long id)".

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]
Great job, I leave you just a few comments

File: src/main/java/com/wolox/training/models/Book.java (hunk 1 above)
Comment (leandrocostawolox) [reply src/main/java/com/wolox/training/models/Book.java:19]: Remember to add the `@Column` annotation in genre :)

**Review by leandrocostawolox — CHANGES_REQUESTED** [review]

**Review by leandrocostawolox — APPROVED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Pull request merged**